src/System.Runtime/tests/System/ValueTypeTests.cs

```diff
@@ -16,10 +16,190 @@ public static void ToStringTest()
             Assert.Equal("System.Tests.ValueTypeTests+S", obj.ToString());
         }
 
```

Comment:
For completeness, it would be nice to have a test for regular cases that used to always work.